src/actions/sources/createPrettySource.js

```diff
@@ -13,6 +13,14 @@ export function createPrettySource(sourceId) {
     const url = getPrettySourceURL(source.get("url"));
     const id = await sourceMaps.generatedToOriginalId(sourceId, url);
 
+    dispatch({
+      type: "UPDATE_SOURCE",
+      source: {
+        id: sourceId,
+        loadedState: "loading"
+      }
+    });
```

Comment:
I was imagining that we would add the pretty source here w/ the loadedState `loading` and below, we would update the pretty source w/ the text and loadedState `loaded`

Reply:
Yes please .. that should take care of #4832